gems/activestorage/6.0/app/models/active_storage/blob.rbs

```diff
@@ -0,0 +1,150 @@
+  # it allows permanent URLs that redirect to the +service_url+ to be cached in the view.
+  def service_url: (?expires_in: untyped expires_in, ?disposition: ::Symbol disposition, ?filename: untyped? filename, **untyped options) -> untyped
+
+  # Returns a URL that can be used to directly upload a file for this blob on the service. This URL is intended to be
+  # short-lived for security and only generated on-demand by the client-side JavaScript responsible for doing the uploading.
+  def service_url_for_direct_upload: (?expires_in: untyped expires_in) -> untyped
+
+  # Returns a Hash of headers for +service_url_for_direct_upload+ requests.
+  def service_headers_for_direct_upload: () -> untyped
+
+  # Uploads the +io+ to the service on the +key+ for this blob. Blobs are intended to be immutable, so you shouldn't be
+  # using this method after a file has already been uploaded to fit with a blob. If you want to create a derivative blob,
+  # you should instead simply create a new blob based on the old one.
+  #
+  # Prior to uploading, we compute the checksum, which is sent to the service for transit integrity validation. If the
+  # checksum does not match what the service receives, an exception will be raised. We also measure the size of the +io+
+  # and store that in +byte_size+ on the blob record. The content type is automatically extracted from the +io+ unless
+  # you specify a +content_type+ and pass +identify+ as false.
+  #
+  # Normally, you do not have to call this method directly at all. Use the +create_and_upload!+ class method instead.
+  # If you do use this method directly, make sure you are using it on a persisted Blob as otherwise another blob's
+  # data might get overwritten on the service.
+  def upload: (untyped io, ?identify: bool identify) -> untyped
+
+  def unfurl: (untyped io, ?identify: bool identify) -> untyped
+
+  def upload_without_unfurling: (untyped io) -> untyped
+
+  # Downloads the file associated with this blob. If no block is given, the entire file is read into memory and returned.
+  # That'll use a lot of RAM for very large files. If a block is given, then the download is streamed and yielded in chunks.
+  def download: () { () -> untyped } -> untyped
+
+  # Downloads the blob to a tempfile on disk. Yields the tempfile.
+  #
+  # The tempfile's name is prefixed with +ActiveStorage-+ and the blob's ID. Its extension matches that of the blob.
+  #
+  # By default, the tempfile is created in <tt>Dir.tmpdir</tt>. Pass +tmpdir:+ to create it in a different directory:
+  #
+  #   blob.open(tmpdir: "/path/to/tmp") do |file|
+  #     # ...
+  #   end
+  #
+  # The tempfile is automatically closed and unlinked after the given block is executed.
+  #
+  # Raises ActiveStorage::IntegrityError if the downloaded data does not match the blob's checksum.
+  def open: (?tmpdir: untyped? tmpdir) { () -> untyped } -> untyped
```

Comment:
`open` yields the block with a tempfile.

```suggestion
  def open: (?tmpdir: untyped? tmpdir) { (Tempfile) -> untyped } -> untyped
```


https://github.com/rails/rails/blob/0d304eae601f085274b2e2c04316e025b443da62/activestorage/app/models/active_storage/blob.rb#L216-L219
https://github.com/rails/rails/blob/0d304eae601f085274b2e2c04316e025b443da62/activestorage/lib/active_storage/service.rb#L86-L88
https://github.com/rails/rails/blob/0d304eae601f085274b2e2c04316e025b443da62/activestorage/lib/active_storage/downloader.rb#L11-L17

Reply:
✨